File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    phone_dict = {}
    
    for num in phone_book:
        phone_dict[num] = True
    
    for num in phone_book:
        temp = ""
        for digit in num[:-1]:
            temp += digit
            if temp in phone_dict:
                return False
    
    return True

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.cpp
#include <string>
#include <vector>
#include <unordered_set>

using namespace std;

bool solution(vector<string> phone_book) {
    unordered_set<string> phone_set;
    
    for (string s : phone_book){
        phone_set.insert(s);
    }
    
    for (string s : phone_book){
        string temp = "";
        for (int i=0;i<s.size()-1;i++){
            temp += s[i];
            if (phone_set.count(temp)){
                return false;
            }
        }
    }
    
    return true;
}